### PR TITLE
feat: get a single newsletter

### PR DESCRIPTION
**Pull request by Okunola11**
I added the get single newsletter endpoint to the route of the project at `/api/v1/pages/newsletters/{id}`. The endpoint accepts the `get` method.

## Description

This endpoint fetches the details of the newsletter with the provided ID. The endpoint is accessible by all. 
## Related Issue (Link to issue ticket)

The issue for this pull request can be found [here](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/762).

## Motivation and Context

This endpoint lets anyone get the details of a newsletter. The request `id` must be of a newsletter which exists in the database. Any other id get a `404` error.

## How Has This Been Tested?

The endpoint has been tested. The unit tests are in the file `tests/v1/newsletter/test_get_single_newsletter.py`.
- [x] Data is fetched appropriately from the database.
- [x] Appropriate response and status code is returned. 
- [x] Anyone can get a newsletter.
- [x] Non existing `id` will throw a `404` error. 

The test can be run by: ` PYTHONPATH=. pytest tests/v1/newsletter/test_get_single_newsletter.py`.

### Affected areas of the code
- I added the required logic to the `newsletter.py` file in the route, service and schema folders. The files outline the model for my response data and logic for the endpoint respectively. 
- I added a test file at `tests/v1/newsletter/test_get_single_newsletter.py`.

### Success Response
Screenshot from 2024-08-07 13-34-59

## Types of changes

- [x] New feature (non-breaking change which adds functionality)    ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the readme accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/newsletter.py [hunk 1]
```diff
@@ -49,3 +50,20 @@ def retrieve_subscribers(
...
+
+    Args:
+        id: The id of the job for the newsletter
+        db: database Session object
+
+    Returns:
+        SingleNewslettersResponse: response on success
+    """
+    newsletterservice = NewsletterService()
+    return newsletterservice.fetch(news_id=id, db=db)
```
Comment (joboy-dev) [inline api/v1/routes/newsletter.py:69]: Where is the success response??

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/routes/newsletter.py (hunk 1 above)
Comment (Okunola11) [reply api/v1/routes/newsletter.py:69]: That's in the service file. The fetch method called returns that

Comment (joboy-dev) [reply api/v1/routes/newsletter.py:75]: Okay

**Review by joboy-dev — APPROVED** [review]

**Pull request merged**